ports/cheriot-rtos/machine_uart.c

```diff
@@ -0,0 +1,205 @@
+        { MP_QSTR_rx, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL} },
+        { MP_QSTR_cts, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL} },
+        { MP_QSTR_rts, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL} },
+        { MP_QSTR_timeout, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 100} },
+        { MP_QSTR_timeout_char, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 100} },
+        { MP_QSTR_invert, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = -1} },
+        { MP_QSTR_flow, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = -1} },
+        { MP_QSTR_txbuf, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 32} },
+        { MP_QSTR_rxbuf, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 64} },
+    };
+    // Parse args.
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    if (args[ARG_bits].u_int != 8) {
+        mp_raise_ValueError("Bit hard coded to 8");
+    }
+
+    if (args[ARG_stop].u_int != 1) {
+        mp_raise_ValueError("Stop bits hard coded to 1");
+    }
+
+    if ((args[ARG_tx].u_obj != MP_OBJ_NULL) ||
+        (args[ARG_rx].u_obj != MP_OBJ_NULL) ||
+        (args[ARG_cts].u_obj != MP_OBJ_NULL) ||
+        (args[ARG_rts].u_obj != MP_OBJ_NULL)) {
+        mp_raise_NotImplementedError(
+            MP_ERROR_TEXT("explicit choice of tx/rx/rts/cts pins is not implemented"));
+    }
+
+    if (args[ARG_txbuf].u_int != 32) {
+        mp_raise_ValueError("tx buffer is 32 bytes long");
+    }
+
+    if (args[ARG_rxbuf].u_int != 64) {
+        mp_raise_ValueError("rx buffer is 64 bytes long");
+    }
+
+    if (args[ARG_invert].u_int != -1) {
+        mp_raise_NotImplementedError("UART invert not implemented");
+    }
+
+    if (args[ARG_flow].u_int != -1) {
+        mp_raise_NotImplementedError("UART flow not implemented");
+    }
+
+    // uint32_t parity = 0;
+    if (args[ARG_parity].u_obj != MP_OBJ_NULL) {
+        mp_raise_NotImplementedError("UART parity not implemented");
+        // parity = mp_obj_get_int(args[ARG_parity].u_obj);
+    }
+
+    self->timeout_ms = args[ARG_timeout].u_int;
+    self->char_timeout_ms = args[ARG_timeout_char].u_int;
+
+    uint32_t baudrate = args[ARG_baudrate].u_int;
+    uart_init(self->uart_block, baudrate);
+}
+
+
+static mp_obj_t mp_machine_uart_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
+    mp_arg_check_num(n_args, n_kw, 1, MP_OBJ_FUN_ARGS_MAX, true);
+
+    int id = mp_obj_get_int(args[0]);
+    volatile open_titan_uart_t *block = get_uart_block(id);
+
+    machine_uart_obj_t *self = m_new_obj(machine_uart_obj_t);
+    self->base.type = &machine_uart_type;
+    self->uart_block = block;
+
+    mp_map_t kw_args;
+    mp_map_init_fixed_table(&kw_args, n_kw, args + n_args);
+    mp_machine_uart_init_helper(self, n_args - 1, args + 1, &kw_args);
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+
+// TODO this one too
```

Comment:
Unresolved TODO?

Reply:
Indeed I forgot to take this one out. I dont think the function actually needs to do anything however...